src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -648,12 +626,34 @@ export function getInputType(
                     Description: getDoc(program, value) ?? "",
                     Type: inputType,
                     IsRequired: !value.optional,
-                    IsReadOnly: isReadOnly,
-                    IsDiscriminator: false
```

Comment:
There are a lot of changes because we do not assign `IsDisriminator` property explicitly to false here. I believe we should keep it, according to the definition of `InputModelProperty`, `IsDiscriminator` property is required for that interface.
This would also reduce the amount of changed files in this PR

Reply:
discussed with dapeng, we'll:
- keep the sequence of discriminator property if it's defined in model
- insert the default discriminator property as the first property, if it's not defined in model